src/metabase/lib/js.cljs

```diff
@@ -386,14 +387,29 @@
   [a-query stage-number a-filter-clause]
   (lib.core/filter-operator a-query stage-number a-filter-clause))
 
-(defn ^:export filter-parts
-  "Returns the parts (operator, args, and optionally, options) of `filter-clause`."
-  [a-query stage-number a-filter-clause]
-  (let [{:keys [operator options column args]} (lib.core/filter-parts a-query stage-number a-filter-clause)]
-    #js {:operator operator
-         :options (clj->js (select-keys options [:case-sensitive :include-current]))
-         :column column
-         :args (to-array args)}))
+(defn ^:export expression-clause
+  "Returns a standalone clause for an `operator`, `options`, and arguments."
+  [an-operator options args]
+  (lib.core/expression-clause an-operator (js->clj options :keywordize-keys true) args))
+
+(defn ^:export expression-parts
+  "Returns the parts (operator, args, and optionally, options) of `expression-clause`."
+  [a-query stage-number an-expression-clause]
+  (let [parts (lib.core/expression-parts a-query stage-number an-expression-clause)]
+    (walk/postwalk
```

Comment:
I guess that a custom expression-parts walking function could be significantly faster then the generic postwalk. This can be ignored for now, of course.